mypy/checkexpr.py

```diff
@@ -2218,6 +2218,8 @@ def bool_type(self) -> Instance:
         return self.named_type('builtins.bool')
 
     def narrow_type_from_binder(self, expr: Expression, known_type: Type) -> Type:
+        if isinstance(known_type, ClassVarType):
+            known_type = known_type.item
```

Comment:
This indeed looks strange, and probably would be not necessary if class variables would be implemented as flags for types or ``Var`` nodes.